stingray/pulse/pulsar.py

```diff
@@ -526,40 +558,70 @@ def fftfit_error(phase, prof, template, p0, **fftfit_kwargs):
     """
     nstep = _default_value_if_no_key(fftfit_kwargs, "nstep", 100)
 
-    approx, _ = _pulse_template(phase, prof)
-    sigma = np.std(prof - approx)
-    nbin = len(prof)
+    if sigma is None:
+        sigma = np.sqrt(np.mean(template))
+
+    nbin = len(template)
 
     ph_fit = np.zeros(nstep)
     amp_fit = np.zeros(nstep)
     # use bootstrap method to calculate errors
     for i in range(nstep):
-        newprof = approx + np.random.normal(0, sigma, nbin)
+        newprof = np.random.normal(0, sigma, len(template)) + template
+        dph = np.random.normal(0, 0.5 / nbin)
+        p0 = [1, dph]
         res = minimize(_fft_fun_wrap, p0, args=([newprof, template],),
                        method='L-BFGS-B',
-                       bounds=[[0, None], [0, None]])
+                       bounds=[[0, None], [-1, 1]],
+                       options={'maxiter': 10000})
+
         amp_fit[i] = res.x[0]
-        ph_fit[i] = res.x[1]
 
-    p0 -= np.floor(p0)
+        ph_fit[i] = normalize_phase_0d5(res.x[1])
 
     std_save = 1e32
     # avoid problems if phase around 0 or 1: shift, calculate std,
     # if less save new std
-    for shift in np.arange(0, 0.8, 0.1):
+    for shift in np.arange(0, 0.8, 0.2):
         phs = ph_fit + shift
         phs -= np.floor(phs)
-        std = np.std(phs)
-        if std <= std_save:
+        std = mad(phs)
+        if std < std_save:
             std_save = std
-            mean_save = np.mean(phs) - shift
+            mean_save = np.median(phs) - shift
 
-    # But still, never less than half a bin!
-    std_save = np.max([std_save, np.diff(phase)[0] / 2])
     return np.mean(amp_fit), np.std(amp_fit), mean_save, std_save
 
 
+def plot_TOA_fit(profile, template, toa, mod=None, toaerr=None,
+                 additional_phase=0., show=True, period=1):
+    """Plot diagnostic information on the TOA."""
```

Comment:
Please add parameter descriptions to docstring.

Reply:
This was meant to be private, used only by get_TOA, so I made it private :)